src/map/unit.cpp

```diff
@@ -2378,8 +2462,13 @@ int32 unit_skilluse_id2(struct block_list *src, int32 target_id, uint16 skill_id
 	if( casttime <= 0 )
 		ud->state.skillcastcancel = 0;
 
-	if (!sd || sd->skillitem != skill_id || skill_get_cast(skill_id, skill_lv))
-		ud->canact_tick = tick + i64max(casttime, max(status_get_amotion(src), battle_config.min_skill_delay_limit)) + SECURITY_CASTTIME;
+	// Skills used from items don't seem to give any attack or act delay
+	if (sd == nullptr || sd->skillitem != skill_id) {
```

Comment:
I think this logic should be moved into unit_set_attackdelay.
The function call should be moved after setting ud->skill_id, then you access it in unit_set_attackdelay

Reply:
Mainly did it like this based on the principle that functions should be kept small. Ideally it is said a function should be fully visible on 1-2 screens.

Sure, I could make unit_set_attackdelay a massive 10 page function that handles everything delay-related, but that doesn't seem to follow good coding standards.

And making it depend on certain variables being set beforehand seems also dangerous.

Reply:
Also please consider that unit_set_attackdelay is designed to be bl specific. Things that should happen independent of bl don't really fit into it (you'd need to make a separate switch/if outside of the bl->type switch to add logic like this there).